### PR TITLE
enabling static themes in prod

### LINKED ISSUE #5799: Turn on static themes flag for production
Set `enableStaticThemes: true` for AMO in production.

### PR DESCRIPTION
**Pull request by championshuttler**
Fixes #5799 

I hope we need this only :see_no_evil:

### CODE REVIEW

**Comment by championshuttler** [comment]
Hi @muffinresearch, should I create another PR for removing the flag from other places as Rebicca created another issue for it or add it to this PR Only ;)

**Comment by muffinresearch** [comment]
@championshuttler the other issue is about removing the use of the code rather than the other config.

This patch should also remove the config where it's set to true for [local dev](https://github.com/mozilla/addons-frontend/blob/master/config/development-amo.js#L13), [dev](https://github.com/mozilla/addons-frontend/blob/3701a9a3afb85a3571435dc97c4f28d9f7a07efa/config/development-amo.js#L13) and [stage](https://github.com/mozilla/addons-frontend/blob/3701a9a3afb85a3571435dc97c4f28d9f7a07efa/config/stage-amo.js#L28).

**Comment by championshuttler** [comment]
Oops sorry i read it wrongly :/ .
@muffinresearch done , ready for another look

**Comment by muffinresearch** [comment]
Looks like there are test failures because some of the tests rely on this configuration being false. Those will need to be updated.

/cc @rebmullin

**Comment by rebmullin** [comment]
@championshuttler, let me know if you run into issues with the tests and I will try to help : )

File: tests/unit/amo/sagas/test_addonsByAuthors.js [hunk 1]
```diff
@@ -50,6 +50,8 @@ describe(__filename, () => {
   }
 
   it('calls the API with addonType if set', async () => {
+    const fakeConfig = getFakeConfig({ enableStaticThemes: false });
+    const root = render({ _config: fakeConfig };
```
Comment (rebmullin) [inline tests/unit/amo/sagas/test_addonsByAuthors.js]: actually in this test case you can remove this and add the following where you see addonType here:

`addonType: getAddonTypeFilter(ADDON_TYPE_THEME)`

you will need to import this function (getAddonTypeFilter) too at the top of the file:
import { getAddonTypeFilter } from 'core/utils';

File: tests/unit/amo/components/TestLandingPage.js [hunk 2]
```diff
@@ -308,9 +308,12 @@ describe(__filename, () => {
     const fakeParams = {
       visibleAddonType: visibleAddonType(ADDON_TYPE_THEME),
     };
+
+    const fakeConfig = getFakeConfig({ enableStaticThemes: false });
+
     const match = { params: fakeParams };
 
-    const root = render({ match });
+    const root = render({ _config: fakeConfig }, { match });
```
Comment (rebmullin) [inline tests/unit/amo/components/TestLandingPage.js]: This one is a little bit more involved. For some reason the filter type is showing as extension. I am not sure why this isnt' failing as is: / Anyway, we should be using the getAndLoadLandingAddons function as it makes it easy to pass in the type and we can use the getAddonTypeFilter to determine which filter to use:

```diff
it('sets the links in each footer for themes', () => {
-  store.dispatch(
-    landingActions.loadLanding({
-      addonType: ADDON_TYPE_THEME,
-      featured: createAddonsApiResult([
-       { ...fakeAddon, name: 'Featured', slug: 'featured' },
-     ]),
-      highlyRated: createAddonsApiResult([
-       { ...fakeAddon, name: 'High', slug: 'high' },
-      ]),
-      trending: createAddonsApiResult([
-        { ...fakeAddon, name: 'Trending', slug: 'trending' },
-      ]),
-    }),
- );

+  const addonType = getAddonTypeFilter(ADDON_TYPE_THEME);

+  _getAndLoadLandingAddons({ addonType });

  const fakeParams = {
    visibleAddonType: visibleAddonType(ADDON_TYPE_THEME),
  };

  const fakeConfig = getFakeConfig({ enableStaticThemes: false });

  const match = { params: fakeParams };

- const root = render({ _config: fakeConfig }, { match });
+ const root = render({ match });

  const addonCards = root.find(LandingAddonsCard);
  expect(addonCards.at(0)).toHaveProp('footerLink', {
    pathname: '/search/',
-    query: { addonType: ADDON_TYPE_THEME, featured: true },
+    query: { addonType, featured: true },
  });
  expect(addonCards.at(1)).toHaveProp('footerLink', {
    pathname: '/search/',
-   query: { addonType: ADDON_TYPE_THEME, sort: SEARCH_SORT_TOP_RATED },
+   query: { addonType, sort: SEARCH_SORT_TOP_RATED },
  });
  expect(addonCards.at(2)).toHaveProp('footerLink', {
    pathname: '/search/',
-   query: { addonType: ADDON_TYPE_THEME, sort: SEARCH_SORT_TRENDING },
+   query: { addonType, sort: SEARCH_SORT_TRENDING },
  });
});

```

also make sure to import the getAddonTypeFilter if it's not already here

**Comment by championshuttler** [comment]
Hi @rebmullin done can you please check why `security/snyk` is failing, i dont have access to check it

**Comment by rebmullin** [comment]
hey @championshuttler, I think if you rebase you shouldn't see that anymore? 

is that right @willdurand?

**Comment by championshuttler** [comment]
Hey, @rebmullin  yes its fixed after rebasing the PR.

Thanks to William :smile:

**Comment by rebmullin** [comment]
Looks 👍 

Thanks @championshuttler : )

**Pull request merged**